stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -354,6 +354,49 @@ function copyto!(A::AbstractSparseMatrixCSC, B::AbstractSparseMatrixCSC)
     return A
 end
 
+copyto!(A::AbstractMatrix, B::AbstractSparseMatrixCSC) = _sparse_copyto!(A, B)
+# Ambiguity resolution
+copyto!(A::PermutedDimsArray, B::AbstractSparseMatrixCSC) = _sparse_copyto!(A, B)
+
+function _sparse_copyto!(dest::AbstractMatrix, src::AbstractSparseMatrixCSC{T}) where {T}
+    dest === src && return dest
+    checkbounds(dest, axes(src)...)
+    fill!(dest, zero(T))
+    @inbounds for col in 1:size(src, 2), ptr in nzrange(src, col)
+        row = rowvals(src)[ptr]
+        val = nonzeros(src)[ptr]
+        dest[row, col] = val
+    end
+    return dest
+end
+
+function copyto!(dest::AbstractMatrix{T}, Rdest::CartesianIndices{2},
+                 src::AbstractSparseMatrixCSC, Rsrc::CartesianIndices{2}) where {T}
+    isempty(Rdest) && return dest
+    if size(Rdest) != size(Rsrc)
+        throw(ArgumentError("source and destination must have same size (got $(size(Rsrc)) and $(size(Rdest)))"))
+    end
+    checkbounds(dest, first(Rdest))
+    checkbounds(dest, last(Rdest))
+    checkbounds(src, first(Rsrc))
+    checkbounds(src, last(Rsrc))
+    src′ = Base.unalias(dest, src)
```

Comment:
This is taken straight from `copyto!` in `Base`. The idea is that if you want to copy a region of `A` into a different region of `A` it will be allowed, but that requires making a copy of `A` as source first. If you did not `unalias` then the copy of overlapping portions of the two regions will be wrong. I will add a test.

Reply:
This is taken from the copyto! in Base. The idea is that copying from region_1 in A to region_2 in the same A is allowed, but in general (when the two regions overlap) it needs to make a copy of A as source to avoid a wrong result.  The `unalias` is a conservative approach: always makes a copy if `src === dest`. One could perhaps  be more clever and do this without a copy, but that is more difficult. I will add a test.